### PR TITLE
Remove moment

### PR DESCRIPTION
**Pull request by RikkiGibson**
Continuation of Azure/ms-rest-js#75

### CODE REVIEW

File: src/vanilla/CodeNamerTS.cs [hunk 1]
```diff
@@ -149,7 +149,7 @@ public override string EscapeDefaultValue(string defaultValue, IModelType type)
                     case KnownPrimaryType.DateTimeRfc1123:
                         return $"new Date('{defaultValue}')";
                     case KnownPrimaryType.TimeSpan:
-                        return $"moment.duration('{defaultValue}')";
+                        return $"'{defaultValue}";
```
Comment (RikkiGibson) [inline src/vanilla/CodeNamerTS.cs]: Missing a single quote here--surprised no test errors. Where is this used?

File: test/azure/generated/AzureCompositeModelClient/operations/primitive.ts [hunk 2]
```diff
@@ -2048,8 +2047,8 @@ export class Primitive {
     let field = (options && options.field !== undefined) ? options.field : undefined;
     // Validate
     try {
-      if(field && !moment.isDuration(field)) {
-        throw new Error('field must be of type moment.duration.');
+      if(field && !msRest.isDuration(field)) {
+        throw new Error('field must be of type string.');
```
Comment (RikkiGibson) [inline test/azure/generated/AzureCompositeModelClient/operations/primitive.ts]: This is what happens when you generalize... There's a template string in the generator `"{0} must be of type {1}"` so we might as well just live with the lousy message here and wait for the new generator to improve it.

File: test/vanilla/acceptanceTests.ts [hunk 3]
```diff
@@ -876,12 +875,9 @@ describe('typescript', function () {
         });
       });
 
-      it('should properly handle invalid value for Duration', function (done) {
+      it.skip('should properly handle invalid value for Duration', function (done) {
```
Comment (RikkiGibson) [inline test/vanilla/acceptanceTests.ts]: We decided against doing validation of duration strings on deserialization, but this test might get re-enabled once we start allowing user preferences to type check during deserialization.

**Comment by azuresdkci1x** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.5.11*

## ❌️ General: 89%

<details><summary>57 out of 535 features are not covered by tests</summary><p>

[`FileStreamNonempty`](https://github.com/Azure/autorest.testserver/search?q=FileStreamNonempty), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpClientFailure400Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Delete), [`HttpClientFailure400Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Get), [`HttpClientFailure400Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Head), [`HttpClientFailure400Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Patch), [`HttpClientFailure400Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Post), [`HttpClientFailure400Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Put), [`HttpClientFailure401Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure401Head), [`HttpClientFailure402Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure402Get), [`HttpClientFailure403Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure403Get), [`HttpClientFailure404Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure404Put), [`HttpClientFailure405Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure405Patch), [`HttpClientFailure406Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure406Post), [`HttpClientFailure407Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure407Delete), [`HttpClientFailure409Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure409Put), [`HttpClientFailure410Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure410Head), [`HttpClientFailure411Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure411Get), [`HttpClientFailure412Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure412Get), [`HttpClientFailure413Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure413Put), [`HttpClientFailure414Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure414Patch), [`HttpClientFailure415Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure415Post), [`HttpClientFailure416Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure416Get), [`HttpClientFailure417Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure417Delete), [`HttpClientFailure429Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure429Head), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`ResponsesScenarioEmptyErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioEmptyErrorBody), [`ResponsesScenarioNoModelErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioNoModelErrorBody), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`additionalPropertiesInProperties`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInProperties), [`additionalPropertiesInPropertiesWithAPTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInPropertiesWithAPTypeString), [`additionalPropertiesTrue`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTrue), [`additionalPropertiesTypeObject`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeObject), [`additionalPropertiesTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeString), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`animalNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=animalNotFoundError), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`expectedNoErrors`](https://github.com/Azure/autorest.testserver/search?q=expectedNoErrors), [`expectedPetHungryError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetHungryError), [`expectedPetSadError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetSadError), [`getDurationInvalid`](https://github.com/Azure/autorest.testserver/search?q=getDurationInvalid), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`intError`](https://github.com/Azure/autorest.testserver/search?q=intError), [`linkNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=linkNotFoundError), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putComplexPolymorphismNoDiscriminator`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismNoDiscriminator), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`stringError`](https://github.com/Azure/autorest.testserver/search?q=stringError), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 99%

<details><summary>1 out of 118 features are not covered by tests</summary><p>

[`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by RikkiGibson** [comment]
The Windows CI build succeeded--for some reason the timing of the merge seems to have prevented this from registering.

**Pull request merged**